nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/RestLookupService.java

```diff
@@ -0,0 +1,435 @@
+        proxyConfigurationService = context.getProperty(PROXY_CONFIGURATION_SERVICE)
+                .asControllerService(ProxyConfigurationService.class);
+
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+
+        setAuthenticator(builder, context);
+
+        if (proxyConfigurationService != null) {
+            setProxy(builder);
+        }
+
+        final SSLContextService sslService = context.getProperty(SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+        final SSLContext sslContext = sslService == null ? null : sslService.createSSLContext(SSLContextService.ClientAuth.WANT);
+        if (sslService != null) {
+            builder.sslSocketFactory(sslContext.getSocketFactory());
+        }
+
+        client = builder.build();
+
+        String path = context.getProperty(RECORD_PATH).isSet() ? context.getProperty(RECORD_PATH).getValue() : null;
+        if (!StringUtils.isBlank(path)) {
+            recordPath = RecordPath.compile(path);
+        }
+
+        recordPathName = context.getProperty(RECORD_PATH_PROPERTY_NAME).isSet()
+            ? context.getProperty(RECORD_PATH_PROPERTY_NAME).evaluateAttributeExpressions().getValue()
+            : null;
+
+        getHeaders(context);
+
+        compiledQuery = Query.prepare(context.getProperty(URL).getValue());
+    }
+
+    @OnDisabled
+    public void onDisabled() {
+        this.compiledQuery = null;
+    }
+
+    private void getHeaders(ConfigurationContext context) {
+        headers = new HashMap<>();
+        for (PropertyDescriptor descriptor : context.getProperties().keySet()) {
+            if (descriptor.isDynamic()) {
+                headers.put(
+                    descriptor.getDisplayName(),
+                    context.getProperty(descriptor).evaluateAttributeExpressions().getValue()
+                );
+            }
+        }
+    }
+
+    private void setProxy(OkHttpClient.Builder builder) {
+        ProxyConfiguration config = proxyConfigurationService.getConfiguration();
+        if (!config.getProxyType().equals(Proxy.Type.DIRECT)) {
+            final Proxy proxy = config.createProxy();
+            builder.proxy(proxy);
+
+            if (config.hasCredential()){
+                builder.proxyAuthenticator((route, response) -> {
+                    final String credential= Credentials.basic(config.getProxyUserName(), config.getProxyUserPassword());
+                    return response.request().newBuilder()
+                            .header("Proxy-Authorization", credential)
+                            .build();
+                });
+            }
+        }
+    }
+
+    @Override
+    public Optional<Record> lookup(Map<String, Object> coordinates) throws LookupFailureException {
+        final String endpoint = determineEndpoint(coordinates);
+        final String mimeType = (String)coordinates.get(MIME_TYPE_KEY);
+        final String method   = (String)coordinates.get(METHOD_KEY);
```

Comment:
A value is required because it's a required key, so I don't think that's necessary.

Reply:
When I used the lookup service from LookupRecord, I felt passing a literal value as RecordPath was a bit tricky. I had to use `concat('get')`.

Reply:
@MikeThomsen RequiredKeys is not mandatory, it can be an empty Set. Also, this is a `Lookup` service. Get should be the default method like the famous cURL command.

Reply:
I can see how that would be awkward. Do you want to handle that with a combination of property + lookup key (latter overriding the former) or defer to fixing `LookupRecord`?

Reply:
I want RestLookupService to have 'get' as default value. It will cover 80% of use-cases I assume. I will send a PR to your PR and if it looks fine, then I'd give my +1 so that you can merge.